db/view/view.cc

```diff
@@ -1364,7 +1364,7 @@ void view_builder::initialize_reader_at_current_token(build_step& step) {
     step.reader = make_local_shard_sstable_reader(
             step.base->schema(),
             std::move(permit),
-            make_lw_shared<sstables::sstable_set>(step.base->get_sstable_set()),
+            step.base->get_sstable_set().get(),
```

Comment:
I'll change it to get_snapshot in the next version, especially because the sstable_set is mostly owned by a lw_shared_ptr in the current version. I've used the initial idea from the suggestion in the issue, but I probably shouldn't expect everyone to read that as well.